### PR TITLE
fix(useMagicKeys): throw error on Vue 2.7

### PR DESCRIPTION
**Pull request by rick-you**
### Description

Fixes https://github.com/vueuse/vueuse/issues/2311

### Additional context

---

### What is the purpose of this pull request? 

- [X] Bug fix
- [ ] New Feature
- [ ] Documentation update
- [ ] Other

### Before submitting the PR, please make sure you do the following

- [X] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [X] Read the [Pull Request Guidelines](https://github.com/vueuse/vueuse/blob/main/packages/guidelines.md).
- [X] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [X] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [X] Ideally, include relevant tests that fail without this PR but pass with it.

### CODE REVIEW

**Review by boomyao — DISMISSED** [review]

File: packages/core/useMagicKeys/index.ts [hunk 1]
```diff
@@ -2,6 +2,7 @@ import type { ComputedRef } from 'vue-demi'
 import { computed, reactive, ref, unref } from 'vue-demi'
 import type { MaybeComputedRef } from '@vueuse/shared'
 import { noop } from '@vueuse/shared'
+import { useReactiveSet } from '@vueuse/shared/utils/set'
```
Comment (antfu) [inline packages/core/useMagicKeys/index.ts:5]:
```suggestion
import { useReactiveSet } from '@vueuse/shared'
```

**Comment by antfu** [comment]
Can you add some tests? This seems to break the demo page.

**Comment by luvsic3** [comment]
I managed to fix the error on useReactiveSet, but not knowing how it is fixed, maybe you can add more detailed comment in useReactiveSet

**Comment by antfu** [comment]
I am not confident about it. Adding this means every user would need to pay the cost (size and computational runtime cost) using this function regardless if they are on 2.7 or 3.0+. I feel the complexity is now worth it.

I think it would be better either:

- Get Vue 2.7 support reactive Map natively
- Explicit says that `useMagicKeys` is not supported with 2.7, and someone can release a fork version doing it.

**Pull request closed without merging**